### PR TITLE
Collapse proposal

### PR DESCRIPTION
**Pull request by louismaximepiton**
Proposal for collapse component to add `.collapsed` after animation. In our mind, it makes sense to add `.collapsed` after the end of the collapsing animation.

Co-authored-by: Nurovek <kevin.lediouron.ext@orange.com>
Co-authored-by: louismaximepiton <louismaxime.piton@orange.com>

### CODE REVIEW

File: js/src/collapse.js [hunk 1]
```diff
@@ -184,21 +184,21 @@ class Collapse extends BaseComponent {
...
-      }
-    }
-
     this._isTransitioning = true
 
     const complete = () => {
       this._isTransitioning = false
       this._element.classList.remove(CLASS_NAME_COLLAPSING)
       this._element.classList.add(CLASS_NAME_COLLAPSE)
       EventHandler.trigger(this._element, EVENT_HIDDEN)
```
Comment (GeoSot) [inline js/src/collapse.js:193]: 'Hidden' event maybe has to be triggered after transferred code snippet.  
Seems more logical to my eyes, as if we listen a hidden event, we suppose that the classes will be handled already. right?

Comment (louismaximepiton) [reply js/src/collapse.js:193]: You're totally right, I'm changing it !

**Comment by XhmikosR** [comment]
Is there any way we could drop the `setTimeout`s in the changed tests?

**Comment by louismaximepiton** [comment]
Hi @XhmikosR, I'm not expert with tests, but I don't think so with this solution, since we're waiting for the transition to finish, to apply `.collapsed`. The transition duration is defined with `$transition-collapse`. However, I don't know if it is possible that the test timeout take into account the `transition-duration`.

**Comment by GeoSot** [comment]
As you transferred the event after the change, some of them can be reverted as they were, and you may use `hide` event too :wink:

**Comment by GeoSot** [comment]
@XhmikosR are we fine now to proceed?

**Comment by XhmikosR** [comment]
Yeah, you go ahead if you think it's fine 👍

**Comment by louismaximepiton** [comment]
Thanks a lot for your help on this topic !

**Comment by XhmikosR** [comment]
Actually, wait, I now see the effect of the change.

I'm not sure if it makes sense:

* https://twbs-bootstrap.netlify.app/docs/5.1/components/accordion/#example
* https://deploy-preview-35915--twbs-bootstrap.netlify.app/docs/5.1/components/accordion/#example

the accordion color now changes after the transition is done. I'm not saying it's wrong per se, but it's a change in behavior. It feels like the transition happens in two steps now, if you catch my drift.

/CC @ffoodd @patrickhlauke for your thoughts.

**Comment by patrickhlauke** [comment]
@XhmikosR I can see what you mean about it now seeming to be a "two-step" process. Wonder if the styling should not just be based on the presence or absence of `.collapsed`, but adding a `.collapsing` class to the button as well (so the button goes from no class, to `.collapsing`, to `.collapsed`, similar to what the panels do, and making the `.accordion-button:not(.collapsed)` styles also include a check for this extra class `.accordion-button:not(.collapsed):not(.collapsing)`

confirming thought that from an accessibility point of view, this change still behaves as it should for screen reader users

**Comment by louismaximepiton** [comment]
Was a bit busy last days, do you want me to try something with `.collapsing` on the `<button>` ?

**Comment by patrickhlauke** [comment]
@louismaximepiton no rush. if you do get a chance, that'd be great though yes. did my rough comment make sense about how i'd suggest tackling it at a high level?

**Comment by patrickhlauke** [comment]
purely from a functional/aesthetic point of view, this looks good to me @louismaximepiton ... bon travail :)

**Comment by mdo** [comment]
I think the deploy preview looks good here. Should we freshen this up for merging?

**Comment by GeoSot** [comment]
As I recently spent too many hours, with the help of @XhmikosR, trying to clean this code, preparing it for next major version, and this change just adds complexity, I would prefer to keep it as it is, and for sure discuss any change for v6. 

The is a draft plan, all the open-close components, to be streamlined as behavior and used classes, and of course, adding the same generic classes during transitions, will help in consistency and will be easier to handle through css.

**Comment by louismaximepiton** [comment]
Hey, thanks a lot for the spent hours. The main idea of this PR was to simplify the CSS work, so if you think it would be easier and you will tackle with better consistency among the diffrent components, let's wait for v6 then 😄

**Comment by GeoSot** [comment]
Thanks a lot for your understanding. Keep in touch for v6. 
Any help will be needed and appreciated

**Comment by louismaximepiton** [comment]
I'll reopen it or maybe open a new PR for this in v6.

**Pull request closed without merging**